corehq/apps/export/models/new.py

```diff
@@ -450,24 +456,54 @@ def _insert_system_properties(cls, export_type, table):
             BOTTOM_MAIN_CASE_TABLE_PROPERTIES,
             CASE_HISTORY_PROPERTIES,
             PARENT_CASE_TABLE_PROPERTIES,
+            STOCK_COLUMN,
         )
+
+        nested_repeat_count = len([node for node in table.path if node.is_repeat])
+        extra_args = {
+            'repeat': nested_repeat_count,  # Used for determining the proper row column
+            'domain': domain,  # Used for the StockExportColumn
+        }
         if export_type == FORM_EXPORT:
             if table.path == MAIN_TABLE:
-                cls.__insert_system_properties(table, TOP_MAIN_FORM_TABLE_PROPERTIES)
-                cls.__insert_system_properties(table, BOTTOM_MAIN_FORM_TABLE_PROPERTIES, top=False)
+                cls.__insert_system_properties(table, TOP_MAIN_FORM_TABLE_PROPERTIES, **extra_args)
+                cls.__insert_system_properties(
+                    table,
+                    BOTTOM_MAIN_FORM_TABLE_PROPERTIES,
+                    top=False,
+                    **extra_args
+                )
             else:
-                cls.__insert_system_properties(table, [ROW_NUMBER_COLUMN])
+                cls.__insert_system_properties(table, [ROW_NUMBER_COLUMN], **extra_args)
         elif export_type == CASE_EXPORT:
             if table.path == MAIN_TABLE:
-                cls.__insert_system_properties(table, TOP_MAIN_CASE_TABLE_PROPERTIES)
-                cls.__insert_system_properties(table, BOTTOM_MAIN_CASE_TABLE_PROPERTIES, top=False)
+                if Domain.get_by_name(domain).commtrack_enabled:
+                    top_properties = TOP_MAIN_CASE_TABLE_PROPERTIES + [STOCK_COLUMN]
+                else:
+                    top_properties = TOP_MAIN_CASE_TABLE_PROPERTIES
+                cls.__insert_system_properties(table, top_properties, **extra_args)
+                cls.__insert_system_properties(
+                    table,
+                    BOTTOM_MAIN_CASE_TABLE_PROPERTIES,
+                    top=False,
+                    **extra_args
+                )
             elif table.path == CASE_HISTORY_TABLE:
-                cls.__insert_system_properties(table, CASE_HISTORY_PROPERTIES)
+                cls.__insert_system_properties(table, CASE_HISTORY_PROPERTIES, **extra_args)
             elif table.path == PARENT_CASE_TABLE:
-                cls.__insert_system_properties(table, PARENT_CASE_TABLE_PROPERTIES)
+                cls.__insert_system_properties(table, PARENT_CASE_TABLE_PROPERTIES, **extra_args)
 
     @classmethod
-    def __insert_system_properties(cls, table, properties, top=True):
+    def __insert_system_properties(cls, table, properties, top=True, **data):
```

Comment:
The `data` arg could use a more descriptive name. Maybe `column_initialization_data` or something like that?